.github/workflows/ci.yml

```diff
@@ -23,6 +23,10 @@ on:
       - main
   workflow_dispatch:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.ref_name }}-${{ github.event.number || github.sha }}
```

Comment:
wouldn't this mean that we are grouping it to the same sha groups vs just PR?

Reply:
good question! the sha group is only used as a fallback in the case that a PR number isn't defined (```github.event.number```) which will be the case when it isn't a PR (push to main and in workflow_dispatch). https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs#example-using-a-fallback-value